src/adapters/output/postgres/customer.ts

```diff
@@ -20,25 +20,26 @@ export class CustomerRepository implements CustomerRepositoryPort {
         createdAt: insertCustomer.raw[0].created_at,
       };
 
-      return result;
+      return result
     } catch (error) {
-      if (error instanceof Error) throw new Error(`Error adding customer: ${error.message}`);
+      if (error instanceof Error)
+        throw new Error(`Error adding customer: ${error.message}`);
       throw new Error(`Error adding customer: ${error}`);
     }
   }
 
-  async searchByCpf(paramCpf: string): Promise<Customer> {
+  async searchByCpf(paramCpf: string): Promise<Customer | string> {
```

Comment:
You could return `null` when the customer is not found instead of returning a msg. WDYT?